### DISCUSSION TITLE
Having trouble with pyfetch.

### DISCUSSION THREAD

**Discussion by S10MC2015** [pyodide/pyodide#2182] [Q&A] [answered]
Hello, 
I am trying to use pyfetch to fetch an image and then it will be downloaded. I am having issues with await and have not a clue how to fix it.

```py
async def pyfetchreq(url):
    pyfetchreqdata = await pyodide.http.pyfetch(getendpoint.format(url = url))
    #logging.debug("Get request (normal) data: " + getreqdata.getvalue())
    logging.debug("pyfetch Get request Grabbing: " + url)
    return pyfetchreqdata

async def coverdl(coverurl):
    with open("cover.jpg",'wb') as f:
        f.write(await pyfetchreq(coverurl).bytes())

await coverdl(shtestbook.coverimageurl)
bookmake.set_cover("cover.jpg",open('cover.jpg', 'rb').read(), create_page=True)

```
(this is with top level async allowed)

The issue I keep having is "<exec>:117: RuntimeWarning: coroutine 'pyfetchreq' was never awaited" which leads to "AttributeError: 'coroutine' object has no attribute 'bytes'"

**Comment by hoodmane** [comment, marked as answer]
Try: `f.write((await pyfetchreq(coverurl)).bytes())`?

**Reply by S10MC2015** [reply]
![image](https://user-images.githubusercontent.com/30575036/154508426-79588a42-1a58-4a06-a7dd-f533ff1f13ad.png)

**Reply by hoodmane** [reply]
Oh yeah you need another `await`:
`f.write(await (await pyfetchreq(coverurl)).bytes())`
